### PR TITLE
add network error parsing

### PR DESCRIPTION
**Pull request by tc-wilson**
added parsing of network errors, took out duplicate digit matching from a regex, removed some printfs for Karl

### CODE REVIEW

File: sysstat-post-process [hunk 1]
```diff
@@ -394,12 +394,54 @@ for my $log_file (sort readdir($dh)) {
...
+                                $desc{'type'} = 'errors-sec';
+				foreach my $direction (keys %networkerrors) # $direction keys are rx and tx
+					{
+					my $error_flavor = $networkerrors{$direction};
+					foreach my $flavor (keys %$error_flavor) # $flavor keys are collision|drop|frame-alignment|carrier|fifo-overrun
+						{
+						$sample{'value'} = $networkerrors{$direction}{$flavor};
+						my %names = ('dev' => $dev, 'direction' => $direction, 'type' => get_netdev_type($dev, $netdev_types_ref), 'error' => $flavor);
+						log_sample($source, \%desc, \%names, \%sample);
+						}
```
Comment (k-rister) [inline sysstat-post-process:433]: the indentation seems all out of whack

File: sysstat-post-process [hunk 2]
```diff
@@ -394,12 +394,54 @@ for my $log_file (sort readdir($dh)) {
...
+                                if (defined $prev_hms_ms and $prev_hms_ms > $hms_ms) {
+                                    # hour-minute-second is lower than last reading so one day has passed
+                                    $ymd_timestamp_ms = advance_ymd($ymd_timestamp_ms);
+                                }
+                                my %sample = ('end' => $ymd_timestamp_ms + $hms_ms);
+                                $desc{'type'} = 'errors-sec';
+				foreach my $direction (keys %networkerrors) # $direction keys are rx and tx
+					{
+					my $error_flavor = $networkerrors{$direction};
+					foreach my $flavor (keys %$error_flavor) # $flavor keys are collision|drop|frame-alignment|carrier|fifo-overrun
```
Comment (k-rister) [inline sysstat-post-process:428]: You can write this more simply as:

```
foreach my $flavor (keys %{$networkerrors{$direction}})
```

**Review by k-rister — CHANGES_REQUESTED** [review]
Its probably fine once the indentation is corrected.

File: sysstat-post-process (hunk 1 above)
Comment (atheurer) [reply sysstat-post-process:433]: Yes, and generally we have the '{' on the same line at the end of the loop or conditional, like:
`foreach my $direction (keys %networkerrors) { # $direction keys are rx and tx`

**Review by k-rister — APPROVED** [review]

**Review by atheurer — CHANGES_REQUESTED** [review]
@tc-wilson once you remove lines 409 and 410 this PR should be fine

**Review by atheurer — COMMENTED** [review]
Am I mistaken, or is there a ton of code that has been changed, that is not related to the net-errors?  There seems to be a ton of code changed with spaces added to contents in ().  Why is that? @tc-wilson @k-rister

**Comment by tc-wilson** [comment]
I ran it through a filter that applies the formatting from the Perl Style Guide, with the exception that I forced the 'elses' to be fully cuddled the way you like them.    ex:   } else {

**Review by atheurer — CHANGES_REQUESTED** [review]
I don't believe the rx/tx packet stats have been added yet? @tc-wilson

**Review by atheurer — APPROVED** [review]

**Pull request merged**